File: 507.cpp
class Solution {
public:
    bool checkPerfectNumber(int num) {
        int i = 2, sum = 1;
        if(num <= 1) {return false;}

        while(i * i < num) {
            if(num % i == 0) {
                sum += i + num/i;
            }
            i++;
        }
        
        if(i * i == num) {
            sum += i;
        }

        return sum == num;
    }
};

File: 507.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        i, total = 1, 0

        while (i * i < num):
            if(num % i == 0):
                total += i + num//i
            i += 1

        if i*i == num:
            total += i

        return total == num * 2